internal/controller/kymaenvironmentbinding/kymaenvironmentbinding.go

```
@@ -0,0 +1,200 @@
+		cr.Status.AtProvider.Bindings = []v1alpha1.Binding{}
+	}
+
+	// Check if we have any valid active bindings
+	hasValidBinding, validBindings := c.validateBindings(cr)
+	cr.Status.AtProvider.Bindings = validBindings
+
+	// If we already have a valid binding, return its details
+	if hasValidBinding {
+		err := c.kube.Status().Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalCreation{}, err
+		}
+	}
+
+	// Create new binding only if we don't have a valid one
+	clientBinding, err := c.client.CreateInstance(ctx, *cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	// Create new binding from client binding
+	newBinding := v1alpha1.Binding{
+		Id:        clientBinding.Metadata.Id,
+		IsActive:  true,
+		CreatedAt: metav1.NewTime(time.Now().UTC()),
+		ExpiresAt: metav1.NewTime(clientBinding.Metadata.ExpiresAt.UTC()),
+	}
+
+	// Add new binding to status
+	cr.Status.AtProvider.Bindings = append(cr.Status.AtProvider.Bindings, newBinding)
+	err = c.kube.Status().Update(ctx, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	// Prepare connection details
+	connectionDetails := managed.ConnectionDetails{
+		"binding_id": []byte(newBinding.Id),
+		"expires_at": []byte(newBinding.ExpiresAt.UTC().String()),
+		"created_at": []byte(newBinding.CreatedAt.UTC().String()),
+		"kubeconfig": []byte(clientBinding.Credentials.Kubeconfig),
+	}
+
+	return managed.ExternalCreation{
+		ConnectionDetails: connectionDetails,
+	}, nil
+}
+
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+
+	return managed.ExternalUpdate{}, errors.New("Update is not implemented - should not be called, only create")
+}
+
+func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1alpha1.KymaEnvironmentBinding)
+	if !ok {
+		return errors.New(errNotKymaEnvironmentBinding)
+	}
+
+	err := c.client.DeleteInstance(ctx, cr)
+	err2 := c.kube.Status().Update(ctx, cr)
```

Why do we update here? I think we should return the Deletion error first if it occurs and not put the update error first.

---

Agree to the faulty error handling.

We have the issue here, that the state "which binding belongs to us" is not on the API side, but on the status side. This status is not too critical, since it all has its TTL anyway, but its still relevant, that in case of deletion we keep the list of bindings up to date. Since an error could occur and we dont delete the relevant bindings. 
This answer applies to all `.Status().Update()` calls. Any better ideas?

---

Updated the procedure to not call `.Status().Update()` in `Delete()`.

It will now call the API in observe and sort out all bindings which have already been deleted.